hdfs-sink/src/main/java/org/springframework/cloud/stream/module/hdfs/sink/HdfsSinkProperties.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.hdfs.sink;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import javax.validation.constraints.Pattern;
+
+/**
+ * Used to configure those Hdfs Sink module options that are related to connecting to Hdfs.
+ *
+ * @author Thomas Risberg
+ */
+@SuppressWarnings("unused")
+@ConfigurationProperties
+public class HdfsSinkProperties {
+
+	/**
+	 * URL for HDFS Namenode
+	 */
+	private String fsUri;
+
+	/**
+	 * base path to write files to
+	 */
+	@Value("/tmp/${spring.application.name:hdfs-sink}")
+	private String directory;
```

Comment:
Should we get directory from a stream definition name so that we'd align how it works in current XD hdfs sink?